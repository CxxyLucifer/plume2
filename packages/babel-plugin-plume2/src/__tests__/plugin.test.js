const plume2 = require('../index.js');
const babel = require('babel-core');

const example = `
import ReactDOM from 'react-dom';
var ReactDOM1 = require('react-dom');`;

describe('babel-plugin-plume2 test suite', () => {
  it('no react-native code', () => {
    const { code } = babel.transform(example, {
      plugins: [plume2]
    });
    expect(code).toEqual(example);
  });

  it('react-native match-snapshot', () => {
    const { code } = babel.transform(example, {
      plugins: [[plume2, { reactnative: true }]]
    });
    expect(code).toMatchSnapshot();
  });

  it('optimize QL name', () => {
    const example = `
    const helloQL = QL('helloQL', [/*...*/]);
    const worldQL = QL('worldQL', [/*...*/]);
    `;
    const { code } = babel.transform(example, {
      env: {
        production: {
          plugins: ['plume2']
        }
      }
    });

    expect(code).toMatchSnapshot();
  });

  it('optimize StoreProvider opts', () => {
    const example = `
    StoreProvider(AppStore, {debug: true})
    `;
    const { code } = babel.transform(example, {
      env: {
        production: {
          plugins: ['plume2']
        }
      }
    });
    expect(code).toMatchSnapshot();
  });
});
