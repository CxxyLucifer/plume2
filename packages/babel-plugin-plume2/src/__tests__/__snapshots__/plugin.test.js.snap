// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-plume2 test suite optimize QL name 1`] = `
"
const helloQL = QL('', [/*...*/]);
const worldQL = QL('', [/*...*/]);"
`;

exports[`babel-plugin-plume2 test suite optimize StoreProvider opts 1`] = `
"
StoreProvider(AppStore);"
`;

exports[`babel-plugin-plume2 test suite react-native match-snapshot 1`] = `
"
import ReactDOM from 'react-native';
var ReactDOM1 = require('react-native');"
`;
